cloud/scope/machine.go

```diff
@@ -367,6 +367,15 @@ func (m *MachineScope) InstanceSpec() *compute.Instance {
 			instance.ShieldedInstanceConfig.EnableIntegrityMonitoring = false
 		}
 	}
+	if m.GCPMachine.Spec.OnHostMaintenance != nil {
+		instance.Scheduling.OnHostMaintenance = strings.ToUpper(string(*m.GCPMachine.Spec.OnHostMaintenance))
```

Comment:
This could become fragile if a new value were ever added, consider a switch for an enum instead
```suggestion
		switch *m.GCPMachine.Spec.OnHostMaintenance {
		case infrav1. HostMaintenancePolicyMigrate:
			instance.Scheduling.OnHostMaintenance = "MIRGATE"
		case infrav1. HostMaintenancePolicyTerminate:
			instance.Scheduling.OnHostMaintenance = "TERMINATE"
		default:
			// Log a warning that an unknown type was met/return an error
		}

		instance.Scheduling.OnHostMaintenance = strings.ToUpper(string(*m.GCPMachine.Spec.OnHostMaintenance))
```

Reply:
There's a webhook that validates the spec but I don't mind updating it.